### PR TITLE
gl: Use factored scale offset function call

### PR DESCRIPTION
**Pull request by vlj**
This PR factors the scale offset matrix computation in gl backend and use uniform buffer for that.

Unfortunatly glm doesn't support uniform buffer so I had to use manual gl call for that.

### CODE REVIEW

File: rpcs3/Emu/RSX/GL/GLGSRender.cpp [hunk 1]
```diff
@@ -1063,6 +1063,11 @@ void GLGSRender::oninit_thread()
 	m_vao.create();
 	m_vbo.create();
 	m_ebo.create();
+	m_scale_offset_buffer.create(16 * sizeof(float));
+
+	PFNGLBINDBUFFERBASEPROC glBindBufferBase = (PFNGLBINDBUFFERBASEPROC)wglGetProcAddress("glBindBufferBase");
```
Comment (tambry) [inline rpcs3/Emu/RSX/GL/GLGSRender.cpp]: wglGetProcAddress is a Windows only function, as the "w" prefix indicates.

**Comment by vlj** [comment]
:/ I forget that.
Can we have a lib that loads function pointer for us ? (like glew) because loading pointer for every platform is tedious.

**Comment by tambry** [comment]
@vlj I wouldn't mind, but the decision will be up to @Nekotekina.

**Comment by danilaml** [comment]
why do we need glm anyway?

**Comment by vlj** [comment]
It provides some wrapper around opengl objects. It provides some matrix maths helpers too but we don't really need them.

**Comment by vlj** [comment]
is the GLProcTable.h a reliable way to add a function pointer ?

**Comment by vlj** [comment]
Rebased + build fixed I think

**Comment by vlj** [comment]
rebased

**Pull request merged**